libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3811,7 +3811,7 @@ void TypeChecker::endVisit(UsingForDirective const& _usingFor)
 				3100_error,
 				path->location(),
 				"The function \"" + joinHumanReadable(path->path(), ".") + "\" "+
-				"cannot be bound to the type \"" + _usingFor.typeName()->annotation().type->humanReadableName() +
+				"cannot be bound to the type \"" + _usingFor.typeName()->annotation().type->canonicalName() +
```

Comment:
This is a tricky one. The problem with `humanReadableName()` is that it includes location, which does not make sense in this context. I.e. when we have `using {f} for S`, the type used currently in the error message is `struct S storage ref`, which makes no sense to the user. Instead we'd like to see `struct S`.

`canonicalName()` is just `S` which is close but could be better. Unfortunately we don't seem to have a variant that prints `struct S` without location so I'm leaving it at that for now.

Reply:
I noticed that `humanReadableName()` when not overridden defaults to `toString(_withoutDataLocation = false)` being called which produces the location names after `struct S`. We can directly call `toString(true)` and then get just `struct S`.
```suggestion
				"cannot be bound to the type \"" + _usingFor.typeName()->annotation().type->toString(true /* withoutDataLocation */) +
```
This seems to also work for other user-defined types `enum` and `contract`.

Reply:
Nice. That's much better. Done.